docs/dev/developer-guides/transactions/blocks.md

```diff
@@ -58,11 +57,51 @@ To avoid this is why we built a purely fictional block containing the event o
+| gasUsed       | The actual amount of gas used in this block.                                                          |
+| transactions  | A list of all transactions included in the block.                                                     |
+| baseFeePerGas | The market price for gas                                                                              |
+
+### Why do we return L2 blocks from API?
+
+On zkSync Era we return L2 blocks from API because this is how all platforms, which include SDKs, Metamask and all other popular wallets can perceive our transactions as processed. It is expected that a transaction is processed once it is included in some block. That's why we need to produce L2 blocks faster than L1 batches.
+### Smart contract
+
+The following are the block properties returned: 
+
+| Parameter     | Description                                                                                           |
+| ------------- | ------------------------------------------------------------------------------------------------------|
+| hash          | The hash of the block.                                                                                |
+| parentHash    | It refers to the hash of the parent block in L1.                                                      |
+| number        | The unique sequential number for this block.                                                          |
+| timestamp     | The current block's creation time in seconds that returns the timestamp of the L1 batch.              |
+| nonce         | The number of transactions sent from a given address                                                  |
+| difficulty    | The current block difficulty returns 2500000000000000 (zkSync does not have proof of work consensus). |
+| gasLimit      | The block gas limit, always returns `2^32-1`.                                                         |
+| gasUsed       | The actual amount of gas used in this block.                                                          |
+| transactions  | A list of all transactions included in the block.                                                     |
+| baseFeePerGas | The market price for gas                                                                              |
+
+### Why do we return L1 batches inside EVM:
+
+In the past,`block.number` was a part of the VM and was provided once per batch and so we returned the number of the batch there.
+Presently, we to return the L2 block inside the contracts, but this value, just like L2 blocks, will be unprovable. We also likely won't have any meaningful value there. Unlike Ethereum blocks that contain some kind of commitment to the state of the chain, on zkSync Era these blocks won't have such commitment, because calculating the Merkle tree is too expensive to be done more often than once per batch.
```

Comment:
We don't. We always return the batch number inside zkEVM. Maybe in the future we will provide some method to retrieve the L2 block number from within the API but not now